src/other/index.jl

```diff
@@ -214,6 +214,27 @@ end
 @inline Base.getindex(x::AbstractIndex, idx::AbstractRange{<:Integer}) =
     getindex(x, collect(Int, idx))
 @inline Base.getindex(x::AbstractIndex, ::Colon) = Base.OneTo(length(x))
+
+@inline Base.getindex(x::AbstractIndex, notidx::Union{Not{<:AbstractVector{Symbol}},
+                                                      Not{<:AbstractVector{<:AbstractString}}}) =
+    setdiff(1:length(x), [getindex(x, idx) for idx in notidx.skip if haskey(x, idx)])
```

Comment:
One more thing I have noted. This approach allows duplicates in `notidx.skip`. Normally we error when duplicates are passed. Maybe we should have `allunique` test on it and throw an error if duplicates are found?